packages/vite/src/node/plugin.ts

```diff
@@ -140,6 +141,12 @@ export interface Plugin<A = any> extends RollupPlugin<A> {
       ctx: HmrContext,
     ) => Array<ModuleNode> | void | Promise<Array<ModuleNode> | void>
   >
+  /**
+   * Use this hook to read and modify bundle and chunk metadata safely.
+   */
+  inheritMetadata?: ObjectHook<
```

Comment:
I wonder instead of this, we just pass it as an arg to the plugin function "constructor"s.

Reply:
Actually wouldn't work for 3rd-party plugins 🤔